include/aspect/material_model/interface.h

```diff
@@ -365,6 +379,31 @@ namespace aspect
        * in order to compute the reaction increment.
        */
       std::vector<std::vector<double> > reaction_terms;
+
+      /**
+       * Vector of shared pointers to additional material model output
+       * objects that can then be added to MaterialModelOutputs. By default,
+       * no outputs are added.
+       */
+      std::vector<std_cxx11::shared_ptr<AdditionalMaterialOutputs<dim> > > additional_outputs;
+
+      /**
+       * Given an additional material model output class, returns a pointer to this
+       * additional material model output object if it used in the current simulation.
+       * The output can then be filled in the evaluate() function. If the output does
+       * not exist, a null pointer is returned.
+       */
+      template <class T>
+      T *get_additional_output()
+      {
+        for (unsigned int i=0; i<additional_outputs.size(); ++i)
+          {
```

Comment:
Please move the implementation of this function out of the class declaration and down to the bottom of the `.h` file.